ouroboros-consensus-test/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -855,13 +876,21 @@ withTestMempool setup@TestSetup {..} prop =
                            , snapshotSlotNo
                            } =
         case runExcept $ repeatedlyM
-               (fmap (TickedSimpleLedgerState . convertMapKind . getTickedSimpleLedgerState  . fst) .: applyTx testLedgerConfig DoNotIntervene snapshotSlotNo)
-               txs
-               (TickedSimpleLedgerState (convertMapKind ledgerState)) of
+               (\tx st -> do
+                   st' <- applyTx testLedgerConfig DoNotIntervene snapshotSlotNo tx (forgetLedgerTablesDiffsTicked st)
+                   pure (zipOverLedgerTablesTicked g (fst st') (projectLedgerTablesTicked st)))
+               [ txForgetValidated tx | (tx, _) <- snapshotTxs ]
+               (mapOverLedgerTablesTicked f $ TickedSimpleLedgerState ledgerState) of
           Right _ -> property True
           Left  e -> counterexample (mkErrMsg e) $ property False
       where
-        txs = map (txForgetValidated . fst) snapshotTxs
+        g :: (Ord k, Eq v) => TrackingMK k v -> TrackingMK k v -> TrackingMK k v
+        g (ApplyTrackingMK vafter dafter) (ApplyTrackingMK _ dbefore) =
+            ApplyTrackingMK vafter (dbefore <> dafter)
+
+        f :: (Ord k, Eq v) => ValuesMK k v -> TrackingMK k v
+        f (ApplyValuesMK v) = ApplyTrackingMK v mempty
```

Comment:
I think this is equivalent to `rawAttacAndApplyDiffs`

Reply:
But we shall rework all the `raw` functions at some point